error/s2n_errno.h

```diff
@@ -335,25 +335,42 @@ extern __thread struct s2n_debug_info _s2n_debug_info;
 #define STRING_(s)     TO_STRING(s)
 #define STRING__LINE__ STRING_(__LINE__)
 
+#define _S2N_RSPLIT(subject, c) (strrchr((subject), c) ? strrchr((subject), c) + 1 : (subject))
+
 /* gets the basename of a file path */
 /* _S2N_EXTRACT_BASENAME("Error encountered in /path/to/my/file.c") -> "file.c" */
 #if !(defined(CBMC) || defined(__TIMING_CONTRACTS__))
-    #define _S2N_RSPLIT(subject, c)     (strrchr((subject), c) ? strrchr((subject), c) + 1 : (subject))
     #define _S2N_EXTRACT_BASENAME(path) _S2N_RSPLIT((path) + strlen(_S2N_DEBUG_LINE_PREFIX), '/')
 #else
     #define _S2N_EXTRACT_BASENAME(path) path
 #endif
 
 #define _S2N_DEBUG_LINE_PREFIX "Error encountered in "
-#define _S2N_DEBUG_LINE        _S2N_DEBUG_LINE_PREFIX __FILE__ ":" STRING__LINE__
 
-#define _S2N_ERROR(x)                                                              \
-    do {                                                                           \
-        _s2n_debug_info.debug_str = _S2N_DEBUG_LINE;                               \
-        _s2n_debug_info.source = _S2N_EXTRACT_BASENAME(_s2n_debug_info.debug_str); \
-        s2n_errno = (x);                                                           \
-        s2n_calculate_stacktrace();                                                \
-    } while (0)
+#ifdef S2N_DEBUG_INFO
+    #define _S2N_DEBUG_LINE _S2N_DEBUG_LINE_PREFIX __FILE__ ":" STRING__LINE__
+#else
+    #define _S2N_DEBUG_LINE _S2N_DEBUG_LINE_PREFIX "s2n-tls. Enable S2N_DEBUG_INFO for more information."
+#endif
```

Comment:
Is the "s2n-tls" part necessary? It seems kind of jarring / missing context. Maybe just the message would be better.
```suggestion
    #define _S2N_DEBUG_LINE _S2N_DEBUG_LINE_PREFIX "Enable S2N_DEBUG_INFO for more information."
#endif
```

Reply:
Well it's concatenating `_S2N_DEBUG_LINE_PREFIX` with that string. So I just put `s2n-tls` instead of the file name.